pkg/cloudprovider/aws/fleet/packing/instancetype.go

```diff
@@ -0,0 +1,104 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package packing
+
+import (
+	"errors"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// TODO get this information from node-selector
+var (
+	nodePools = []*nodeCapacity{
+		{
+			instanceType: "m5.8xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("32000m"),
+				v1.ResourceMemory: resource.MustParse("128Gi"),
+			},
+			utilizedCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.Quantity{},
+				v1.ResourceMemory: resource.Quantity{},
+			},
+		},
+		{
+			instanceType: "m5.2xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("8000m"),
+				v1.ResourceMemory: resource.MustParse("32Gi"),
+			},
+			utilizedCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.Quantity{},
+				v1.ResourceMemory: resource.Quantity{},
+			},
+		},
+		{
+			instanceType: "m5.xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("4000m"),
+				v1.ResourceMemory: resource.MustParse("16Gi"),
+			},
+			utilizedCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.Quantity{},
+				v1.ResourceMemory: resource.Quantity{},
+			},
+		},
+		{
+			instanceType: "m5.large",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("2000m"),
+				v1.ResourceMemory: resource.MustParse("8Gi"),
+			},
+			utilizedCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.Quantity{},
+				v1.ResourceMemory: resource.Quantity{},
+			},
+		},
+	}
+)
+
+var ErrInsufficientCapacity = errors.New("insufficient capacity")
+
+func (p *packingEstimator) getInstanceTypes(filter string) []*nodeCapacity {
+	return nodePools
```

Comment:
Seems like this should return a deep copy of `nodePools`? Otherwise it'll get overwritten over time?

Reply:
This will change very soon most likely in the next PR. We are only updating the available capacity and it gets reset in `podsPackingForInstance` method.